gbpservice/nfp/orchestrator/drivers/orchestration_driver.py

```diff
@@ -375,27 +536,45 @@ def create_network_function_device(self, device_data,
                          "service flavor field, using default "
                          "flavor: m1.medium"))
             flavor = 'm1.medium'
+
         interfaces_to_attach = []
+        advance_sharing_interfaces = []
         try:
             for interface in interfaces:
                 port_id = network_handler.get_port_id(token, interface['id'])
                 interfaces_to_attach.append({'port': port_id})
 
             if not self.supports_hotplug:
-                for port in device_data['ports']:
-                    if port['port_classification'] == nfp_constants.PROVIDER:
-                        port_id = network_handler.get_port_id(
-                                                        token, port['id'])
+                if self.setup_mode.get(nfp_constants.NEUTRON_MODE):
+                    # TODO(ashu): get neutron mode from conf
```

Comment:
COAL is to achieve not using any check for mode in orchestration driver.
network_handler will be set with either gbp_network_driver or neutron_network_driver
the specific implementations should be there in the drivers with method name say, create_ports_for_sharing()